src/elements/content-uploader/ContentUploader.js

```diff
@@ -385,17 +390,21 @@ class ContentUploader extends Component<Props, State> {
 
         const folderItems = [];
         const fileItems = [];
+        const packageItems = [];
 
         Array.from(dataTransferItems).forEach(item => {
             const isDirectory = isDataTransferItemAFolder(item);
-            if (isDirectory && isFolderUploadEnabled) {
+            if (Browser.isSafari() && isDataTransferItemAPackage(item)) {
+                packageItems.push(item);
+            } else if (isDirectory && isFolderUploadEnabled) {
                 folderItems.push(item);
             } else if (!isDirectory) {
                 fileItems.push(item);
             }
         });
 
         this.addFileDataTransferItemsToUploadQueue(fileItems, itemUpdateCallback);
+        if (packageItems.length) this.addPackageDataTransferItemsToUploadQueue(packageItems, itemUpdateCallback);
```

Comment:
We don't check for length before calling the other `add*DataTransferItemsToUploadQueue`, do we need to here?

Reply:
If I had time, I would like to revisit this section. Currently there are some bugs in how this is handled, I think. I noticed that the lists never get cleared, so it ends up with duplicates of previously added data transfer items. I added this `if` to make the debugging less noisy.

I can take this one out, but I do wonder if the better change is to clean the queues and prevent duplicates. Will save that for another day.